lib/restHttp/src/main/scala/com/gu/util/resthttp/Types.scala

```diff
@@ -74,12 +73,14 @@ object Types {
           f.andThen(_.toDisjunction)
 
         val boundAsDisjunction: ClientDisjunction[B] =
-          disjunctionMonad.bind(originalAsDisjunction)(functionWithResultAsDisjunction)
+          disjunctionMonad.flatMap(originalAsDisjunction)(functionWithResultAsDisjunction)
 
         boundAsDisjunction.toClientFailableOp
       }
 
-      override def point[A](a: => A): ClientFailableOp[A] = ClientSuccess(a)
+      override def tailRecM[A, B](a: A)(f: A => ClientFailableOp[Either[A, B]]): ClientFailableOp[B] = ???
```

Comment:
Just trying to be clever:
```suggestion
      @tailrec
      override def tailRecM[A, B](a: A)(f: A => ClientFailableOp[Either[A, B]]): ClientFailableOp[B] = {
        f(a) match {
          case ClientSuccess(Left(newA)) => this.tailRecM(newA)(f)
          case ClientSuccess(Right(newB)) => ClientSuccess[B](newB)
          case failed: ClientFailure => failed
        }
      }
```